From 9a7cc21ad37052cbb36ab02207bfddb9b017ebe0 Mon Sep 17 00:00:00 2001
From: Tram Nguyen <tram.nguyen.xw@rvc.renesas.com>
Date: Thu, 8 Sep 2016 16:19:54 +0700
Subject: [PATCH 3/3] gst145-v4l2src color conversion: Use memcpy_unalign
 instead of memcpy when copy data

Porting from a patch in Linux 3.10 environment

Signed-off-by: Tram Nguyen <tram.nguyen.xw@rvc.renesas.com>
---
 sys/v4l2/Makefile.am  | 1 +
 sys/v4l2/gstv4l2src.c | 4 +++-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/sys/v4l2/Makefile.am b/sys/v4l2/Makefile.am
index 5ccea1a..9817415 100644
--- a/sys/v4l2/Makefile.am
+++ b/sys/v4l2/Makefile.am
@@ -33,6 +33,7 @@ libgstvideo4linux2_la_LIBTOOLFLAGS = $(GST_PLUGIN_LIBTOOLFLAGS)
 libgstvideo4linux2_la_LIBADD =   $(GST_PLUGINS_BASE_LIBS) \
				 -lgstallocators-$(GST_API_VERSION) \
				 -lgstvideo-$(GST_API_VERSION) \
+                                 -lmemcpy \
				 $(GST_BASE_LIBS) \
				 $(GST_LIBS) \
				 $(LIBV4L2_LIBS) \
diff --git a/sys/v4l2/gstv4l2src.c b/sys/v4l2/gstv4l2src.c
index eee3e69..3c207c7 100644
--- a/sys/v4l2/gstv4l2src.c
+++ b/sys/v4l2/gstv4l2src.c
@@ -54,6 +54,8 @@

 #include "gstv4l2src.h"

+#include "memcpy_unalign.h"
+
 #include "gstv4l2colorbalance.h"
 #include "gstv4l2tuner.h"
 #include "gstv4l2vidorient.h"
@@ -795,7 +797,7 @@ gst_v4l2src_create (GstPushSrc * src, GstBuffer ** buf)
     ptr_tmp += vin_width * vin_height;

     for (i=1; i*2 < vin_height; i++)
-        memcpy((ptr_tmp + i*vin_width), (ptr_tmp + i*vin_width*2), vin_width);
+      memcpy_unalign((ptr_tmp + i*vin_width), (ptr_tmp + i*vin_width*2), vin_width);

     size_new = map.size *3/4;
     gst_buffer_unmap (buf, &map);
--
1.9.1
